app/components/TopicPicker/index.js

```diff
@@ -1,58 +1,102 @@
 import React from 'react';
 import { fetchTopic } from 'actions/article';
 import { connect } from 'react-redux';
-
+import { activateTopic,passTopic } from 'actions/article';
+var Radium = require('radium');
 import { styles } from './styles.scss';
 
 const mapDispatchToProps = dispatch => {
   return {
     onActivateTopic: (topic) => {
       dispatch(activateTopic(topic));
-    }
+    },
   };
 }
 
-const mapStateToProps = state => {
+/*const mapStateToProps = state => {
+  console.log('TOPICPICKER STATE')
+  console.log(state)
   return {
-    topic: state.currentTopic
+    currentTopic: state.currentTopic
   };
-}
+}*/
 
 const TopicPicker = React.createClass({
   displayName: 'TopicPicker',
 
   propTypes: {
     topics: React.PropTypes.array.isRequired,
+    onActivateTopic: React.PropTypes.func,
+    currentTopic: React.PropTypes.string.isRequired,
+    //colors: React.PropTypes.array.isRequired
+  },
+
+  componentDidMount() {
+      this.activateTopic('1');
   },
 
-  activateTopic(topicId, props) {
-    console.log(topicId);
+  activateTopic(topic) {
+    this.props.onActivateTopic(topic);
+    //this.props.onPassTopic(topic);
   },
 
   render() {
-    let topics = this.props.topics;
+    var colors =this.props.colors;
+    var index = -1;
+    var defcolor;
+    console.log('TOPICPICKER RENDER')
+    console.log(this.props)
+
+    var list = this.props.topics.map(topic => {
+      var topic_id = topic.id + ''
+      var func = this.activateTopic.bind(this, topic_id);
+      index += 1;
+
+      defcolor = '#4c4545';
+
+      if (topic_id == this.props.currentTopic){
+        defcolor = colors[index]
+      }
+      var styles = {
```

Comment:
Fair enough. I couldn't find any other way of dynamically generating css for hovering. I completely agree that it's really gross. If you find another solution it would definitely be appreciated (and I would gladly delete this section).